controlplane/internal/controller/clusterdeployment_controller.go

```diff
@@ -295,38 +303,87 @@ type InstallConfigOverride struct {
-func setACICapabilities(oacp *controlplanev1alpha2.OpenshiftAssistedControlPlane, aci *hiveext.AgentClusterInstall) error {
-	isBaremetalPlatform := aci.Spec.PlatformType == hiveext.BareMetalPlatformType
-	if isCapabilitiesEmpty(oacp.Spec.Config.Capabilities) && !isBaremetalPlatform {
-		return nil
+// isBaremetalPlatform checks if the AgentClusterInstall is configured for a baremetal platform.
+// Returns true if the platform type is BareMetalPlatformType, false otherwise.
+func isBaremetalPlatform(aci *hiveext.AgentClusterInstall) bool {
+	return aci.Spec.PlatformType == hiveext.BareMetalPlatformType
+}
+
+// getInstallConfigOverride merges install config override from annotations with capabilities-based overrides.
+// It returns the final merged install config override as a JSON string, or empty string if no overrides are needed.
+// The function combines user-provided install config overrides from annotations with automatically
+// generated capabilities configuration based on the cluster platform type.
+func getInstallConfigOverride(oacp *controlplanev1alpha2.OpenshiftAssistedControlPlane, aci *hiveext.AgentClusterInstall) (string, error) {
+	var installConfigOverrideStr string
+	cfg, hasInstallConfigOverride := oacp.Annotations[controlplanev1alpha2.InstallConfigOverrideAnnotation]
+	if hasInstallConfigOverride {
+		installConfigOverrideStr = cfg
+	}
+	capabilitiesCfgOverride, err := getInstallConfigOverrideForCapabilities(oacp, aci)
+	if err != nil {
+		return "", err
+	}
+	// if no override and no capabilities, no need to merge install config override
+	if installConfigOverrideStr == "" && capabilitiesCfgOverride == "" {
+		return "", nil
+	}
+	if capabilitiesCfgOverride == "" {
+		return installConfigOverrideStr, nil
+	}
+	if installConfigOverrideStr == "" {
+		return capabilitiesCfgOverride, nil
+	}
+
+	installConfigOverrideStr, err = mergeJson(installConfigOverrideStr, capabilitiesCfgOverride)
+	if err != nil {
+		return "", err
 	}
+	return installConfigOverrideStr, nil
+}
+
+// getInstallConfigOverrideForCapabilities generates install config override JSON for OpenShift capabilities configuration.
+// It automatically sets appropriate capabilities based on the platform type:
+// - Baremetal platforms: Sets baseline to "None" and includes default baremetal capabilities
+// - Non-baremetal platforms: Sets baseline to "vCurrent" and only includes user-specified capabilities
+// Returns empty string if no capabilities configuration is needed (non-baremetal with empty capabilities).
+func getInstallConfigOverrideForCapabilities(oacp *controlplanev1alpha2.OpenshiftAssistedControlPlane, aci *hiveext.AgentClusterInstall) (string, error) {
 	var installCfgOverride InstallConfigOverride
-	baselineCapability, err := getBaselineCapability(oacp.Spec.Config.Capabilities.BaselineCapability, isBaremetalPlatform)
+	if isCapabilitiesEmpty(oacp.Spec.Config.Capabilities) && !isBaremetalPlatform(aci) {
```

Comment:
Wouldn't this be either the capabilities is empty or it's not baremetal platform? 

```suggestion
	if isCapabilitiesEmpty(oacp.Spec.Config.Capabilities) || !isBaremetalPlatform(aci) {
```

Reply:
I've preserved current logic: if no capablities and not baremetal, nothing to do and return

otherwise we'll call getBaselineCabality (passing wether it's baremetal or not, and everything should be handled in there).

Am I missing something?